drivers/ephy.c

```diff
@@ -383,48 +435,74 @@ int ephy_init(eth_phy_state_t *phy, char *conf, uint8_t board_rev, link_state_cb
 
 	err = mdio_setup(phy->bus, 2500 /* kHz */, 10 /* ns */, 0 /* with-preamble */);
 	if (err != 0) {
+		ephy_printf(phy, "Couldn't init MDIO: %s (%d)", strerror(err), err);
 		return err;
 	}
 
 	ephy_reset(phy);
 
 	phyid = ephy_readPhyId(phy);
 	if (phyid == 0u || phyid == ~0u) {
+		ephy_printf(phy, "Couldn't read PHY ID");
 		gpio_set(&phy->reset, 1);
 		return -ENODEV;
 	}
 
 	/* make address 0 not broadcast, disable NAND-tree mode */
 	ephy_reg_write(phy, EPHY_16_OP_MODE_STRAP_OVERRIDE, (1 << 1) | (1 << 9));
 
-	ephy_setAltConfig(phy, 0); /* KSZ8081 RND - default config */
+	err = ephy_setAltConfig(phy, 0); /* KSZ8081 RND - default config */
+	if (err <= 0) {
+		ephy_printf(phy, "Couldn't set default config");
+		return -ENODEV;
+	}
 
 #ifdef LWIP_EPHY_INIT_HOOK
 	LWIP_EPHY_INIT_HOOK(phy, phyid, board_rev);
+#else
+	(void)board_rev;
 #endif
 
+
 #if defined(EPHY_KSZ8081RNA) || defined(EPHY_KSZ8081RNB)
-	ephy_setAltConfig(phy, 1);
+	cfg_id = 1;
 #elif defined(EPHY_KSZ8081RND)
-	ephy_setAltConfig(phy, 0);
+	cfg_id = 0;
 #endif
+	ephy_setAltConfig(phy, cfg_id);
+	if (err <= 0) {
+		ephy_printf(phy, "Couldn't set alternative config %d", cfg_id);
+		return -ENODEV;
+	}
 
 	phy->link_state_callback = cb;
 	phy->link_state_callback_arg = cb_arg;
-	ephy_show_link_state(phy);
+	ephy_setLinkState(phy);
 
 	if (gpio_valid(&phy->irq_gpio)) {
-		err = beginthread(ephy_link_thread, 0, phy->th_stack, sizeof(phy->th_stack), phy);
+		err = beginthread(ephy_link_thread, 6, phy->th_stack, sizeof(phy->th_stack), phy);
 		if (err != 0) {
 			gpio_set(&phy->reset, 1);
 			return err;
 		}
 
 		/* enable link up/down IRQ signal */
 		ephy_reg_write(phy, EPHY_1B_IRQ_CTRL_STATUS, (1 << 8) | (1 << 10));
+
+		if ((phy->irq_gpio.flags & GPIO_INVERTED) == 0) {
```

Comment:
it's about setting the ephy registers based on provided config, shouldn't it be here? Maybe I should add a function `gpio_inverted` like with `gpio_valid`?

Reply:
why do you want to reconfigure the MACPHY instead of waiting for high / low level in `gpio_wait` depending of the  `GPIO_INVERTED` flag (as it's done for imx6ull?)

Reply:
I see what you mean